src/Nettle.jl

```diff
@@ -23,10 +26,11 @@ end
 
 function __init__()
     global const nettle_major_version = get_libnettle_version()
-    hash_init()
-    cipher_init()
 end
 
-# similar to Python's hmac.HMAC.hexdigest
-hexdigest!(state::Union(HMACState,HashState)) = bytes2hex(digest!(state))
+if VERSION >= v"0.4.0-dev+6521"
+    include("precompile.jl")
+    _precompile_()
```

Comment:
You should put the `__precompile__()` call at the top of the file, before the `module` statement, to avoid double-parsing the module when it is precompiled.

Note that `precompile()` is supported in 0.3, so it doesn't need to be in a `VERSION` check.